### ISSUE TITLE
About implementation of noise normalization term

### ISSUE THREAD

**Issue by julightzhong10** [zeke-xie/Positive-Negative-Momentum#1]
Hi, 

Firstly, thanks for the great work and the code. 

I have some questions on the difference of the "noise normalization term" between released implementation and the paper description. 

1. In the paper and code, as inside the red and blue box, the noise normalization term is calculated by using beta0 which I think is beta3 in the code. But in the code, this term is calculated by using beta2. I don't think there is a big difference since beta2 is 0.999 and beta3 is 1.0, but it might be different if you want to tune the betas.  And I think it might be useful to check if this will effect the experimental results in paper (Although I think it may only lead negligible difference.) 
![paper_shot](https://user-images.githubusercontent.com/39087182/124250052-1079fd80-db57-11eb-81b3-03099ab60825.png)
![code_shot](https://user-images.githubusercontent.com/39087182/124250674-add53180-db57-11eb-961d-f57d5146dca6.jpg)

2. I think this term is a constant during the optimization as long as the betas do not change. So is this acceptable to pre-calculate before optimization start? Or it could be change during optimization? Just want verify my understanding of this algorithm.

Thanks for the reply.

**Comment by zeke-xie** [comment]
Hi julightzhong10,

Thanks for your comments.

1. 
That is a bug in the current open-source implementation of AdaPNM. 
It should be beta3. I will correct it as soon.

Thanks a lot for pointing out this problem.

2. 
Yes, it is fine to calculate the normalization factor directly before optimization. 
This normalization factor is constant during training.

This normalization factor in my implementation can make the optimal hyperparameter settings of PNM/AdaPNM similar to SGD/Adam.

**Comment by julightzhong10** [comment]
Thanks for the reply, close this issue.
